app/controllers/ops_controller/settings/label_tag_mapping.rb

```diff
@@ -52,9 +52,16 @@ def label_tag_mapping_edit
 
   def entity_ui_name_or_all(entity)
     if entity
-      entity = entity.split('::').last
-      entity = 'VmOrTemplate' if entity == 'Image'
-      ui_lookup(:model => entity)
+      provider, model = entity.split('::')
+
+      case model
+      when 'Vm'
+        model = "ManageIQ::Providers::#{provider}::CloudManager::Vm"
+      when 'Image'
+        model = "ManageIQ::Providers::#{provider}::CloudManager::Template"
```

Comment:
@cben Yes, unfortunately. This will have to be refactored again. But, at least we know where everything is and what needs to be updated going forward.